ls1intum/Artemis
# `General`: Improve formatting for long durations

### Checklist
#### General


#### Client

### Motivation and Context
* Generally, long durations were shown as, e.g., `2d 0h 4min 0s`. The updated format removes the unnecessary zero-valued parts to show it as `2d 4min` (unless the duration is zero, then `0s` is still shown).
* On the submissions list pages the durations were always shown in minutes. However, e.g., `11272 minutes` is not intuitive. Those durations are now shown in the format as described above with days/hours/minutes.

### Description
The duration from seconds pipe has been updated, so the change is consistent for all durations across Artemis that use it.

The submissions list views have been updated to now use this pipe as well. There currently is a lot of duplication for those. Fixing this is out-of-scope here, issue #5941 has been created for that. 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 non-quiz Exercise that has a few submissions

1. Log in to Artemis
2. Go to the exercise page.
3. Go to ‘Submissions’. A table with the submissions in this exercise is shown. The duration column should be formatted as described in the Motivation section.

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage




| Class/File | Branch | Line |
|------------|-------:|-----:|
| artemis-duration-from-seconds-pipe | 97% | 97% |


### Screenshots
